src/uwtools/config/validator.py

```diff
@@ -20,85 +20,94 @@
 # Public functions
 
 
-def bundle(schema: dict) -> dict:
+def bundle(schema: dict, keys: Optional[list] = None) -> dict:
     """
     Bundle a schema by dereferencing links to other schemas.
 
     :param schema: A JSON Schema.
+    :param keys: Keys leading up to this block. Internal use only, do not manually specify.
     :returns: The bundled schema.
     """
-    key = "$ref"
+    ref = "$ref"
     bundled = {}
     for k, v in schema.items():
+        newkeys = [*(keys or []), k]
+        key_path = ".".join(newkeys)
         if isinstance(v, dict):
-            if list(v.keys()) == [key] and v[key].startswith("urn:uwtools:"):
+            if list(v.keys()) == [ref] and v[ref].startswith("urn:uwtools:"):
                 # i.e. the current key's value is of the form: {"$ref": "urn:uwtools:.*"}
-                bundled[k] = bundle(_registry().get_or_retrieve(v[key]).value.contents)
+                uri = v[ref]
+                log.debug("Bundling referenced schema %s at key path: %s", uri, key_path)
```

Comment:
Added some debug log messages here to trace the bundling process, which may be helpful if/when something doesn't work as expected.